src/report/user-report.service.ts

```diff
@@ -0,0 +1,35 @@
+import { getRepository } from 'typeorm';
+import { UserReport, CreateReportDto, UpdateStatusDto } from '.';
+
+export const ReportService = new (class {
+    async getAllReports() {
+        const reportsRepo = getRepository(UserReport);
+
+        const reports = await reportsRepo.find();
+
+        if (!reports.length) {
+            return null;
+        }
+        return reports;
+    }
+
+    async createReport(report: CreateReportDto) {
+        const reportsRepo = getRepository(UserReport);
+
+        return reportsRepo.save(report);
```

Comment:
I just realized how bad that is. You cannot do this. It is a serious security hole, because you can specify a certain id in request body. You need to pass each every value by hand, so:

change report argument name to data, and:
```ts
const report = new UserReport();

report.reason = data.reason;

await reportsRepo.save(report);
```